File: Challenge/FIBONACCI/fibonacci.cpp
#include <iostream>

void fibonacci(int n)
{
  unsigned long long fib_sequence[n];
  fib_sequence[0] = 0;
  fib_sequence[1] = 1;

  for (int i = 2; i < n; i++)
  {
    fib_sequence[i] = fib_sequence[i - 1] + fib_sequence[i - 2];
  }

  // Imprimir los primeros 50 números de Fibonacci
  for (int i = 0; i < n; i++)
  {
    std::cout << fib_sequence[i] << " ";
  }
}

int main()
{
  fibonacci(50);
  return 0;
}

File: Challenge/FIBONACCI/fibonacci.rs
fn fibonacci(n: usize) -> Vec<u64> {
    let mut fib_sequence: Vec<u64> = vec![0, 1];

    while fib_sequence.len() < n {
        let next_number =
            fib_sequence[fib_sequence.len() - 1] + fib_sequence[fib_sequence.len() - 2];
        fib_sequence.push(next_number);
    }

    fib_sequence.truncate(n);
    fib_sequence
}

fn main() {
    // Imprimir los primeros 50 números de Fibonacci
    let primeros_50_fibonacci = fibonacci(50);
    println!("{:?}", primeros_50_fibonacci);
}
